### PR TITLE
 Migrate: UDP prospector the input interface

### PR DESCRIPTION
**Pull request by ph**
Refactor the input prospector to use the input interface

Require: #6078

### CODE REVIEW

File: filebeat/prospector/prospector.go [hunk 1]
```diff
@@ -10,6 +10,7 @@ type Prospectorer = input.Input
 
 // Runner encapsulate the lifecycle of a prospectorer
 // Deprecated: See input.Runner
+type Prospector = input.Runner
```
Comment (ruflin) [inline filebeat/prospector/prospector.go]: How is this related to this change?

**Comment by ph** [comment]
@ruflin update/rebased.
